lib/concurrent/promise.rb

```diff
@@ -438,6 +438,46 @@ def self.any?(*promises)
       aggregate(:any?, *promises)
     end
 
+    # Aggregates a collection of promises, and completes as soon as the first
+    # aggregated promise completes, with the same state (fulfilled or rejected)
+    # and reason/value.
+    #
+    #   @param [Array] promises Zero or more promises to aggregate
+    #   @return [Promise] an unscheduled (not executed) promise that aggregates
+    #     the promises given as arguments
+    def self.race(*promises)
+      return Promise.fulfilled if promises.empty?
+
+      Promise.new do
+        latch = Concurrent::CountDownLatch.new(1)
+        finished = Concurrent::AtomicBoolean.new(false)
```

Comment:
Looking at latch, it doesn't seem to use atomic integers under the hood, so this actually might not be possible with the current implementation. I don't won't to derail this PR, but I will take a look at making it use atomics under the hood.

As for using latch, the reason I think it is beneficial is that it uses one datum to ensure synchronization, thus creating a linearization point for the program; removing the possibility of certain race conditions.